packages/openfeature-node-provider/src/index.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  ErrorCode,
+  EvaluationContext,
+  JsonValue,
+  OpenFeatureEventEmitter,
+  Paradigm,
+  Provider,
+  ResolutionDetails,
+  ServerProviderStatus,
+  StandardResolutionReasons,
+} from "@openfeature/server-sdk";
+
+import {
+  BucketClient,
+  ClientOptions,
+  Context as BucketContext,
+} from "@bucketco/node-sdk";
+
+type ProviderOptions = ClientOptions & {
+  contextTranslator?: (context: EvaluationContext) => BucketContext;
+};
+
+const defaultTranslator = (context: EvaluationContext): BucketContext => {
+  const userId = context.targetingKey ?? context["id"];
+  const user = userId
+    ? {
+        id: String(userId),
+        name: context["name"]?.toString(),
+        email: context["email"]?.toString(),
+        country: context["country"]?.toString(),
+      }
+    : undefined;
+
+  const company = context["companyId"]
+    ? {
+        id: String(context["companyId"]),
+        name: context["companyName"],
+      }
+    : undefined;
+
+  return {
+    user,
+    company,
+  };
+};
+
+export class BucketNodeProvider implements Provider {
```

Comment:
Should we add some jsdocs?

Reply:
We could, but you'll realistically be operating against the OpenFeature facade/interface rather than this client directly, so I don't think it's super important.